### PR TITLE
fix: Add Type column to workspace feeds

### PR DESCRIPTION
**Pull request by nkdengineer**
### Fixed Issues

$ https://github.com/Expensify/App/issues/51847
PROPOSAL: https://github.com/Expensify/App/issues/51847#issuecomment-2451699529

### Tests

1. Go to workspace with Expensify card enabled
2. Create a card
3. Verify that: we have Type column
- [x] Verify that no errors appear in the JS console

### Offline tests

same as above
### QA Steps

same as above
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

Screenshot 2024-11-04 at 18 11 12

</details>

<details>
<summary>iOS: mWeb Safari</summary>

Screenshot 2024-11-04 at 18 12 16

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Screenshot 2024-11-04 at 18 07 36

</details>

<details>
<summary>MacOS: Desktop</summary>

Screenshot 2024-11-04 at 18 07 52

</details>

### CODE REVIEW

**Comment by nkdengineer** [comment]
I am a little stuck on the Android builds, will add the screenshots later @DylanDylann

**Comment by mountiny** [comment]
@DylanDylann what is your eta for this checklist? thanks!

File: src/pages/workspace/expensifyCard/WorkspaceExpensifyCardListPage.tsx [hunk 1]
```diff
@@ -100,6 +100,7 @@ function WorkspaceExpensifyCardListPage({route, cardsList}: WorkspaceExpensifyCa
                         limit={item.nameValuePairs?.unapprovedExpenseLimit ?? 0}
                         name={item.nameValuePairs?.cardTitle ?? ''}
                         currency={policyCurrency}
+                        isVirtual={!!item.nameValuePairs?.isVirtual}
```
Comment (DylanDylann) [inline src/pages/workspace/expensifyCard/WorkspaceExpensifyCardListPage.tsx:103]:
```suggestion
                        isVirtual={!!item?.nameValuePairs?.isVirtual}
```

File: src/pages/workspace/expensifyCard/WorkspaceCardListRow.tsx [hunk 2]
```diff
@@ -54,10 +59,15 @@ function WorkspaceCardListRow({limit, cardholder, lastFourPAN, name, currency}:
                         numberOfLines={1}
                         style={[styles.textLabelSupporting, styles.lh16]}
                     >
-                        {name}
+                        {shouldUseNarrowLayout ? `${name} • ${cardType}` : name}
                     </Text>
                 </View>
             </View>
+            {!shouldUseNarrowLayout && (
+                <View style={[styles.flexRow, styles.gap2, shouldUseNarrowLayout ? styles.flex2 : styles.flex1, styles.alignItemsCenter, styles.justifyContentEnd]}>
```
Comment (DylanDylann) [inline src/pages/workspace/expensifyCard/WorkspaceCardListRow.tsx:67]:
```suggestion
                <View style={[styles.flexRow, styles.gap2, styles.flex1, styles.alignItemsCenter, styles.justifyContentEnd]}>
```

Comment (DylanDylann) [reply src/pages/workspace/expensifyCard/WorkspaceCardListRow.tsx:67]: shouldUseNarrowLayout always be false in this case

File: src/pages/workspace/expensifyCard/WorkspaceCardListRow.tsx [hunk 3]
```diff
@@ -54,10 +59,15 @@ function WorkspaceCardListRow({limit, cardholder, lastFourPAN, name, currency}:
...
+                    <Text style={[styles.textLabelSupporting, styles.lh16]}>{cardType}</Text>
```
Comment (DylanDylann) [inline src/pages/workspace/expensifyCard/WorkspaceCardListRow.tsx:68]: please add numberOfLines={1} as other places

**Comment by DylanDylann** [comment]
@nkdengineer Minor comment. The rest looks fine

**Comment by DylanDylann** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>
Screenshot 2024-11-05 at 12 14 07



</details>

<details>
<summary>Android: mWeb Chrome</summary>


Screenshot 2024-11-05 at 12 15 45


</details>

<details>
<summary>iOS: Native</summary>
Screenshot 2024-11-05 at 12 14 11



</details>

<details>
<summary>iOS: mWeb Safari</summary>
Screenshot 2024-11-05 at 12 14 45



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>
Screenshot 2024-11-05 at 12 13 22



</details>

<details>
<summary>MacOS: Desktop</summary>


Screenshot 2024-11-05 at 12 13 41


</details>

**Review by DylanDylann — APPROVED** [review]

**Comment by mountiny** [comment]
@JmillsExpensify @Expensify/design would like to get your input on this one. On smaller screens the plan was to use the dot operator in the card name, but it ends up not visible based on the screenshots. Curious for your take on how to handle the small screens. Even if we add more room to the name column it might be cut off

**Comment by shawnborton** [comment]
We could play with the text styles and try something like this:
CleanShot 2024-11-05 at 10 14 25@2x

The idea above is basically:
- on desktop, just use normal text styles for the table data except the card holder/card name
- on mobile, reuse this same idea but make the card name use micro size, and then put the virtual/physical type under the limit and use micro for that too

cc @Expensify/design for thoughts too

**Comment by mountiny** [comment]
I love that, goes well with the assignee name and card name

**Comment by DylanDylann** [comment]
Kindly bump @Expensify/design

**Comment by dannymcclain** [comment]
Very into [Shawn's idea](https://github.com/Expensify/App/pull/51946#issuecomment-2456635251)!

**Comment by mountiny** [comment]
Should we wait for @dubielzyk-expensify too? I feel like @nkdengineer could start working on the proposed design there

**Comment by dannymcclain** [comment]
I think Jon will agree, but if we want to wait just in case that's fine.

**Comment by mountiny** [comment]
@nkdengineer feel free to implement the change from Shawn

**Comment by nkdengineer** [comment]
I'll update soon

**Comment by DylanDylann** [comment]
@nkdengineer Could you give an ETA for the next update?

**Comment by nkdengineer** [comment]
Working on it

**Comment by nkdengineer** [comment]
web

small

@DylanDylann  i updated, please check again

**Comment by shawnborton** [comment]
Close! But I think we need to update some of the text styles. These selected items here should use our normal font style (15px, regular) and default font color:
CleanShot 2024-11-06 at 10 54 27@2x

**Comment by DylanDylann** [comment]
@nkdengineer Bump on the new design update

**Comment by nkdengineer** [comment]
web

small

@DylanDylann @shawnborton i updated

**Comment by shawnborton** [comment]
Close. In this screenshot here:
CleanShot 2024-11-07 at 09 09 36@2x

The word "Physical" should match the styling of "Hidden's Card"

Please check the [mocks](https://github.com/Expensify/App/pull/51946#issuecomment-2456635251) again for reference!

**Comment by nkdengineer** [comment]
Screenshot 2024-11-07 at 16 14 27

@shawnborton i updated

**Comment by shawnborton** [comment]
That feels better, thanks!

**Review by mountiny — APPROVED** [review]
Thanks for updating the design, seems like we are on a same page now. @nkdengineer before merging, can you also please confirm how it looks like on the wide view to ensure we did not break that when fixing the narrow view? thanks

**Comment by DylanDylann** [comment]
@nkdengineer Please also fix text style on the wide screen

Screenshot 2024-11-07 at 20 57 32

**Comment by nkdengineer** [comment]
@DylanDylann I think it is expected as designed [here](https://github.com/Expensify/App/pull/51946#issuecomment-2459168700)

**Review by DylanDylann — APPROVED** [review]

**Comment by mountiny** [comment]
Yeah that is expected, thanks guys!

**Comment by mountiny** [comment]
Yeah that is expected, thanks guys!

**Comment by mountiny** [comment]
Yeah that is expected, thanks guys!

**Pull request merged**